src/neuroconv/datainterfaces/ecephys/openephys/openephysroutingdatainterface.py

```diff
@@ -0,0 +1,57 @@
+from abc import ABC
+from pathlib import Path
+from typing import Optional
+
+from .openephysdatainterface import OpenEphysBinaryRecordingInterface
+from .openephyslegacydatainterface import OpenEphysLegacyRecordingInterface
+from ....utils import FolderPathType
+
+
+class OpenEphysRecordingInterface(ABC):
+    """Abstract class that defines which interface class to use for a given Open Ephys recording."""
+
+    def __new__(
+        cls,
+        folder_path: FolderPathType,
+        interface_kwargs: Optional[dict] = None,
```

Comment:
This is an interesting approach... normally, this would be `**interface_kwargs` at the top level just as it is when it passes them off after routing

Reply:
Actually, yeah, I'd prefer that... Otherwise the user has to say `interface_kwargs=dict(their_intended_kwargs)` which is not as seamless as just taking and passing them along

The idea being that the `OpenEphysRecordingInterface` should behave exactly like either interface as far as the users can tell, but it figures out which version of the format to use under the hood

Reply:
This is going to be tricky to get a good json schema for because we are not exposing the form of the input params. Would it be possible to homogenize the input args of the two different classes? What is the purpose of having one as "stream_id" and the other as  "experiment_id"?

Reply:
Well, that too... good point on the schema, I forgot that would have problems with that approach anyway

After looking into the [extractor](https://github.com/SpikeInterface/spikeinterface/blob/master/spikeinterface/extractors/neoextractors/openephys.py#L92-L95), the `experiment_id` has a similar role to multi-block, which SI ought to be able to handle as multi-segment, so we can safely ignore that argument here

I can't find a `recording_id` argument either

So our keywords for this interface are `folder_path`, `stream_name`, and that's it actually. No need to handle or pass anything else from what I see

Reply:
I see why this was incorrect, the original intention was to propagate all the arguments that the two interface would need. I changed it now to take `folder_path` and `stream_name`.